cmake: OPENDDS_JUST_BUILD_HOST_TOOLS=TRUE failed

Could it be the `OPENDDS_JUST_BUILD_HOST_TOOLS` in cmake is broken?

@see: https://opendds.readthedocs.io/en/latest-release/devguide/building/index.html#cmake-var-OPENDDS_JUST_BUILD_HOST_TOOLS

I have it at master and at the latest release branch.

```
jan@jan:~/projects$ git clone https://github.com/OpenDDS/OpenDDS.git
Cloning into 'OpenDDS'...
remote: Enumerating objects: 262583, done.
remote: Counting objects: 100% (4924/4924), done.
remote: Compressing objects: 100% (2177/2177), done.
remote: Total 262583 (delta 3361), reused 4189 (delta 2734), pack-reused 257659
Receiving objects: 100% (262583/262583), 121.29 MiB | 8.31 MiB/s, done.
Resolving deltas: 100% (204793/204793), done.
jan@jan:~/projects$ cd OpenDDS/
jan@jan:~/projects/OpenDDS$ cmake -B build -DOPENDDS_JUST_BUILD_HOST_TOOLS=TRUE -DBUILD_SHARED_LIBS=FALSE
-- The CXX compiler identification is GNU 12.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Getting ACE/TAO from https://github.com/DOCGroup/ACE_TAO/releases/download/ACE%2BTAO-7_1_3/ACE%2BTAO-src-7.1.3.tar.bz2
-- Using OpenDDS 3.29.0 at /home/jan/projects/OpenDDS
-- Using MPC at /home/jan/projects/OpenDDS/build/ace_tao/MPC
-- Using ACE 7.1.3 at /home/jan/projects/OpenDDS/build/ace_tao
-- Using TAO 3.1.3 at /home/jan/projects/OpenDDS/build/ace_tao/TAO
-- Found Perl: /usr/bin/perl (found version "5.36.0") 
'/usr/bin/perl' '/home/jan/projects/OpenDDS/cmake/configure_ace_tao.pl' '--mpc' '/home/jan/projects/OpenDDS/build/ace_tao/MPC' '--mpc-type' 'gnuace' '--src' '/home/jan/projects/OpenDDS/build/ace_tao' '--ace' '/home/jan/projects/OpenDDS/build/ace_tao' '--tao' '/home/jan/projects/OpenDDS/build/ace_tao/TAO' '--platform-macros-file' 'platform_linux.GNU' '--workspace-file=/home/jan/projects/OpenDDS/build/host-tools.mwc' '--static=1' '--compiler' '/usr/bin/c++' '--config-file' 'config-linux.h' 'dds_suppress_anys=1' 'no_opendds_safety_profile=1' 'dds_non_coverage=1' 'versioned_namespace=0' 'debug=1' 'optimize=0' 'inline=1' 'static=1' 'xerces3=0' 'ipv6=0' 'no_cxx11=0' 'tao_no_iiop=0' 'tao_optimize_collocated_invocations=1'
workspace-file: /home/jan/projects/OpenDDS/build/host-tools.mwc
compiler: /usr/bin/c++
tao: /home/jan/projects/OpenDDS/build/ace_tao/TAO
static: 1
mpc: /home/jan/projects/OpenDDS/build/ace_tao/MPC
platform-macros-file: platform_linux.GNU
ace: /home/jan/projects/OpenDDS/build/ace_tao
src: /home/jan/projects/OpenDDS/build/ace_tao
config-file: config-linux.h
mpc-type: gnuace
dds_suppress_anys=1
no_opendds_safety_profile=1
dds_non_coverage=1
versioned_namespace=0
debug=1
optimize=0
inline=1
static=1
xerces3=0
ipv6=0
no_cxx11=0
tao_no_iiop=0
tao_optimize_collocated_invocations=1
In "/home/jan/projects/OpenDDS/build/ace_tao" running (list):
 - "/usr/bin/perl"
 - "/home/jan/projects/OpenDDS/build/ace_tao/bin/mwc.pl"
 - "/home/jan/projects/OpenDDS/build/ace_tao/ACE_TAO_for_OpenDDS.mwc"
 - "-type"
 - "gnuace"
 - "-static"
MPC_ROOT was set to /home/jan/projects/OpenDDS/build/ace_tao/MPC.
Using .../build/ace_tao/bin/MakeProjectCreator/config/MPC.cfg
Generating 'gnuace' output using ACE_TAO_for_OpenDDS.mwc
Skipping TAO_IDL_GEN (tao_idl_fe.mpc); it requires tao_idl_fe_gen.
Generation Time: 1s
'/usr/bin/perl' '/home/jan/projects/OpenDDS/cmake/scrape_gnuace.pl' '--workspace' '/home/jan/projects/OpenDDS/build/ace_tao' '--loc-base' '/home/jan/projects/OpenDDS/build' '--ace' '/home/jan/projects/OpenDDS/build/ace_tao' '--tao' '/home/jan/projects/OpenDDS/build/ace_tao/TAO'
--   The C compiler identification is GNU 12.2.0
--   Detecting C compiler ABI info
--   Detecting C compiler ABI info - done
--   Check for working C compiler: /usr/bin/cc - skipped
--   Detecting C compile features
--   Detecting C compile features - done
--   Performing Test CMAKE_HAVE_LIBC_PTHREAD
--   Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
--   Found Threads: TRUE  
-- Configuring done
CMake Error at /usr/share/cmake-3.25/Modules/ExternalProject.cmake:2475:EVAL:2 (add_custom_command):
  Error evaluating generator expression:

    $<TARGET_PROPERTY:OpenDDS_Dcps,CXX_STANDARD>

  Target "OpenDDS_Dcps" not found.
Call Stack (most recent call first):
  /usr/share/cmake-3.25/Modules/ExternalProject.cmake:2475 (cmake_language)
  /usr/share/cmake-3.25/Modules/ExternalProject.cmake:3783:EVAL:2 (ExternalProject_Add_Step)
  /usr/share/cmake-3.25/Modules/ExternalProject.cmake:3783 (cmake_language)
  /usr/share/cmake-3.25/Modules/ExternalProject.cmake:4189 (_ep_add_build_command)
  cmake/build_ace_tao.cmake:90 (ExternalProject_Add)
  CMakeLists.txt:47 (include)


CMake Error at /usr/share/cmake-3.25/Modules/ExternalProject.cmake:2475:EVAL:2 (add_custom_command):
  Error evaluating generator expression:

    $<TARGET_PROPERTY:OpenDDS_Dcps,CXX_STANDARD>

  Target "OpenDDS_Dcps" not found.
Call Stack (most recent call first):
  /usr/share/cmake-3.25/Modules/ExternalProject.cmake:2475 (cmake_language)
  /usr/share/cmake-3.25/Modules/ExternalProject.cmake:3783:EVAL:2 (ExternalProject_Add_Step)
  /usr/share/cmake-3.25/Modules/ExternalProject.cmake:3783 (cmake_language)
  /usr/share/cmake-3.25/Modules/ExternalProject.cmake:4189 (_ep_add_build_command)
  cmake/build_ace_tao.cmake:90 (ExternalProject_Add)
  CMakeLists.txt:47 (include)


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

--- a/.github/workflows/cmake.yml
+++ b/.github/workflows/cmake.yml
@@ -82,6 +82,10 @@ jobs:
         path: OpenDDS/build/ACE_TAO
     - name: Install xerces
       run: sudo apt-get -y install libxerces-c-dev
+    - uses: lukka/get-cmake@latest
+      with:
+        cmakeVersion: "3.23"
+        ninjaVersion: "latest"
     - name: Configure
       run: |
         cd OpenDDS
@@ -170,7 +174,7 @@ jobs:
         path: OpenDDS/build/ACE_TAO
     - uses: lukka/get-cmake@latest
       with:
-        cmakeVersion: "3.23"
+        cmakeVersion: "latest"
         ninjaVersion: "latest"
     - name: Configure
       run: |
@@ -217,7 +221,7 @@ jobs:
         add-to-path: false
     - uses: lukka/get-cmake@latest
       with:
-        cmakeVersion: "3.23"
+        cmakeVersion: "3.24"
         ninjaVersion: "latest"
     - name: Configure Host Tools
       run: |

--- a/.github/workflows/cmake.yml
+++ b/.github/workflows/cmake.yml
@@ -168,8 +168,10 @@ jobs:
       with:
         repository: DOCGroup/ACE_TAO
         path: OpenDDS/build/ACE_TAO
-    - name: Install Ninja
-      run: sudo apt-get -y install ninja-build
+    - uses: lukka/get-cmake@latest
+      with:
+        cmakeVersion: "3.23"
+        ninjaVersion: "latest"
     - name: Configure
       run: |
         cd OpenDDS
@@ -213,8 +215,10 @@ jobs:
       with:
         ndk-version: r25c
         add-to-path: false
-    - name: Install Ninja
-      run: sudo apt-get -y install ninja-build
+    - uses: lukka/get-cmake@latest
+      with:
+        cmakeVersion: "3.23"
+        ninjaVersion: "latest"
     - name: Configure Host Tools
       run: |
         cd OpenDDS
@@ -315,6 +319,10 @@ jobs:
       with:
         vcpkgGitCommitId: '${{ env.VCPKG_GIT_COMMIT }}'
         runVcpkgInstall: true
+    - uses: lukka/get-cmake@latest
+      with:
+        cmakeVersion: "3.23"
+        ninjaVersion: "latest"
     - name: Checkout OpenDDS
       uses: actions/checkout@v4
       with:

--- a/cmake/build_ace_tao.cmake
+++ b/cmake/build_ace_tao.cmake
@@ -36,11 +36,12 @@ execute_process(
 
 # Get the C++ standard OpenDDS is going to be built with. We are going to force
 # the ACE/TAO build to use the same standard.
-set(_opendds_idl_std "$<TARGET_PROPERTY:opendds_idl,CXX_STANDARD>")
-if(NOT OPENDDS_JUST_BUILD_HOST_TOOLS)
-  set(_opendds_dcps_std "$<TARGET_PROPERTY:OpenDDS_Dcps,CXX_STANDARD>")
+if(TARGET OpenDDS_Dcps)
+  set(_opendds_std_target OpenDDS_Dcps)
+else()
+  set(_opendds_std_target opendds_idl)
 endif()
-set(_opendds_std "$<IF:$<TARGET_EXISTS:OpenDDS_Dcps>,${_opendds_dcps_std},${_opendds_idl_std}>")
+set(_opendds_std "$<TARGET_PROPERTY:${_opendds_std_target},CXX_STANDARD>")
 
 set(_build_cmd "${CMAKE_COMMAND}" -E env "ACE_ROOT=${OPENDDS_ACE}" "TAO_ROOT=${OPENDDS_TAO}")
 if(_OPENDDS_XERCES3_FOR_ACE)

--- a/cmake/build_ace_tao.cmake
+++ b/cmake/build_ace_tao.cmake
@@ -37,7 +37,9 @@ execute_process(
 # Get the C++ standard OpenDDS is going to be built with. We are going to force
 # the ACE/TAO build to use the same standard.
 set(_opendds_idl_std "$<TARGET_PROPERTY:opendds_idl,CXX_STANDARD>")
-set(_opendds_dcps_std "$<TARGET_PROPERTY:OpenDDS_Dcps,CXX_STANDARD>")
+if(NOT OPENDDS_JUST_BUILD_HOST_TOOLS)
+  set(_opendds_dcps_std "$<TARGET_PROPERTY:OpenDDS_Dcps,CXX_STANDARD>")
+endif()
 set(_opendds_std "$<IF:$<TARGET_EXISTS:OpenDDS_Dcps>,${_opendds_dcps_std},${_opendds_idl_std}>")
 
 set(_build_cmd "${CMAKE_COMMAND}" -E env "ACE_ROOT=${OPENDDS_ACE}" "TAO_ROOT=${OPENDDS_TAO}")

--- a/cmake/build_ace_tao.cmake
+++ b/cmake/build_ace_tao.cmake
@@ -1,5 +1,6 @@
 include(ExternalProject)
 
+set(_opendds_std_target OpenDDS_Dcps)
 if(OPENDDS_JUST_BUILD_HOST_TOOLS)
   set(ws "${OPENDDS_BUILD_DIR}/host-tools.mwc")
   file(WRITE "${ws}"
@@ -10,7 +11,12 @@ if(OPENDDS_JUST_BUILD_HOST_TOOLS)
     "}\n"
   )
   list(APPEND _OPENDDS_CONFIGURE_ACE_TAO_ARGS "--workspace-file=${ws}")
+  set(_opendds_std_target opendds_idl)
 endif()
+
+# Get the C++ standard OpenDDS is going to be built with. We are going to force
+# the ACE/TAO build to use the same standard.
+set(_opendds_std "$<TARGET_PROPERTY:${_opendds_std_target},CXX_STANDARD>")
 
 find_package(Perl REQUIRED)
 if(OPENDDS_STATIC)
@@ -33,15 +39,6 @@ execute_process(
   COMMAND_ECHO STDOUT
   COMMAND_ERROR_IS_FATAL ANY
 )
-
-# Get the C++ standard OpenDDS is going to be built with. We are going to force
-# the ACE/TAO build to use the same standard.
-if(TARGET OpenDDS_Dcps)
-  set(_opendds_std_target OpenDDS_Dcps)
-else()
-  set(_opendds_std_target opendds_idl)
-endif()
-set(_opendds_std "$<TARGET_PROPERTY:${_opendds_std_target},CXX_STANDARD>")
 
 set(_build_cmd "${CMAKE_COMMAND}" -E env "ACE_ROOT=${OPENDDS_ACE}" "TAO_ROOT=${OPENDDS_TAO}")
 if(_OPENDDS_XERCES3_FOR_ACE)

--- a/docs/news.d/cmake_just_build_host_tools.rst
+++ b/docs/news.d/cmake_just_build_host_tools.rst
@@ -1,0 +1,7 @@
+.. news-prs: 4646
+
+.. news-start-section: Platform Support and Dependencies
+.. news-start-section: Building with CMake
+- Fixed :ghissue:`configure error <4645>` when using :cmake:var:`OPENDDS_JUST_BUILD_HOST_TOOLS` with CMake <3.28.
+.. news-end-section
+.. news-end-section

